app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from functools import wraps
+
+
+def cache(func: callable) -> callable:
```

Comment:
Import `Callable` from `typing` to annotate `func` and return type here. `Callable` has not been deprecated in the last Python versions.